File: Programmers/체육복/taxijjang/체육복.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<queue>
#include<string>
#include<set>
#include<map>
#include<cstring>
#include<functional>
#include<cmath>

#define SIZE 40
const int INF = 2000000000;

using namespace std;

typedef long long int ll;

int check[SIZE];
int solution(int n, vector<int> lost, vector<int> reserve) {

	for (int i = 0; i < SIZE; i++) {
		check[i] = 1;
	}

	for (auto l : lost) {
		check[l]--;
	}
	for (auto r : reserve) {
		check[r]++;
	}

	int answer = 0;

	for (int i = 1; i <= n; i++) {
		if (check[i] == 2) {
			if (check[i - 1] == 0) {
				check[i]--; check[i + 1]++;
			}
			else if (check[i + 1] == 0) {
				check[i]--; check[i + 1]++;
			}
		}
	}

	for (int i = 1; i <= n; i++) {
		if (check[i])
			answer++;
	}
	return answer;
}

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

	//int n; cin >> n;
	vector<int> lost = { 2,4 };
	vector<int> reserve = { 3 };

	cout << solution(5, lost, reserve);

}

File: Programmers/체육복/Hong/체육복.java
package com.company;

import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        int[] check = new int[n+1]; //체육복 갯수

        Arrays.sort(lost);
        Arrays.sort(reserve);

        for(int i=0; i<reserve.length; i++){
            check[reserve[i]] = 2;
        }

        for(int i=0; i<lost.length; i++){
            if(check[lost[i]] == 2){
                check[lost[i]] = 1;
                answer++;
            }
        }

        for(int i = 0; i<lost.length; i++){
            if(check[lost[i]]==1){
                continue;
            }

            if(lost[i]-1 >= 1 && check[lost[i]-1]==2){
                check[lost[i]-1] = 1;
                answer++;
            }
            else if(lost[i] + 1 <= n && check[lost[i]+1]==2){
                check[lost[i]+1] = 1;
                answer++;
            }
        }

        return answer;
    }
}

class Main {

    public static void main(String[] args) {
        int n = 5;
        int[] lost = {1,2,3};
        int[] reserve = {2,3,4};

        Solution solution = new Solution();
        System.out.println(solution.solution(n,lost,reserve));
    }
}